lib/flight_job/models/script.rb

```diff
@@ -222,6 +241,17 @@ def render
     def render_and_save
       content = render
 
+      # Ensures it claims the ID
+      # NOTE: As this is done after validation, it may trigger a race condition
+      #       This could cause the command to fail, whilst the other succeeds
+      #       Subsequent commands will detect the ID has been taken
+      begin
+        FileUtils.mkdir_p FlightJob.config.scripts_dir
+        FileUtils.mkdir File.expand_path(id, FlightJob.config.scripts_dir)
+      rescue Errno::EEXIST
+        raise InternalError, "Unexpectedly failed to claim ID: #{id}"
+      end
+
```

Comment:
I don't think this works as it should.  If the id has already been taken the `render` method will raise an error due to `valid?(:render)` returning false.  The error message presented to the user is `flight job: Unexpectedly failed to render the script!` instead of `flight job: Unexpectedly failed to claim ID: #{id}`.